### PR TITLE
Translate dish titles

### PR DESCRIPTION
**Pull request by Philipp000**
This is an approach on the translation of the dish titles using [LibreTranslate](https://libretranslate.com/)

Here, every dish title gets translated by sending a request to LibreTranslate using the [LibreTranslateAPI](https://github.com/argosopentech/LibreTranslate-py) package.
As there is a request for each dish title, this is quite slow.

Additionally, there are rate limits on the public APIs and I'm not sure how to handle this properly.
One option is to use [ArgosTranslate](https://github.com/argosopentech/argos-translate/) the algorithm behind LibreTranslate. However, this will require about 100MB of storage for installing a language and I'm not sure if that is a good option.

Target for issue #22

### CODE REVIEW

**Comment by COM8** [comment]
Since 500.000 characters might not be enough for us, could you please have a look at how many chars we have to translate if we would like to parse all dishes for all canteens once?

For the CI we could use [Encrypted Secrets](https://docs.github.com/en/actions/security-guides/encrypted-secrets) and pass the API key as an env variable or as an argument.
There we probably should first check if the "SOME_API_KEY_EAT_API" env variable exists and in case it does, use it. If a user also provides an API-Key via the command line interface, this one should be preferred.

**Comment by Philipp000** [comment]
Okay thanks, i will take a look at the Encrypted Secretes! 

I checked how much chars are there currently, and for all canteens once this is at the moment roughly 60k.
For example mensa-garching is about 250 chars per day for the menu. For one month its around 5k chars. 
So i think the 500k free plan from DeepL should be sufficient, as long as every menu gets translated only once.

Additionally,  dishes are often the same for multiple canteens. However, as far as I understand the scripts, its always run once per canteen? Therefore, sharing translations between canteens would require parsing existing files? Which I would say is bit too much effort, for the translations.

So I would implement the DeepL translations without a lookup/reuse of existing translations?

**Comment by COM8** [comment]
I'd say lets try it with DeepL.
But we have to reduce the amount of CI runs (for the translated results) for this. 
https://github.com/TUM-Dev/eat-api/blob/06a63f454e3e80998bb57e7184f28bdc44019ae7/.github/workflows/publish.yml#L5-L6

**Comment by Philipp000** [comment]
I have now implemented the DeepL translation in my branch.
I also added an additional GitHub Action for the translations to be generated once per week, so there should be no problem regarding the character limit of the free DeepL API.
Therefore, the API_Key needs to be provided via the GitHub secret: DEEPL_API_KEY

I tested the Action once in my repository, and it took about 15mins, which i would say should be okay for a run once per week?

Additionally, I also tested the deployment in my fork. In the gh-pages branch there is an additional `en/` folder, which has the same structure as the root directory and contains the translated dishes.

Do you think there is something more to add/change?

File: src/util.py [hunk 1]
```diff
@@ -24,3 +30,33 @@ def make_duplicates_unique(names_with_duplicates):
...
+    :return: Whether translation was successful
+    """
+    # get api key from environment, abort if not given
+    deepl_api_key = os.environ.get("DEEPL_API_KEY_EAT_API")
+    if deepl_api_key is None:
+        raise Exception("For translation please provide a DeepL api key via DEEPL_API_KEY_EAT_API")
+
+    translator = deepl.Translator(deepl_api_key)
+    # source language is always german
+    source_language = "de"
```
Comment (COM8) [inline src/util.py:50]:
```suggestion
    source_language = "DE"
```

File: src/util.py [hunk 2]
```diff
@@ -24,3 +30,33 @@ def make_duplicates_unique(names_with_duplicates):
...
+
+    # don't use deepl, when already correct language
+    if source_language == language:
```
Comment (COM8) [inline src/util.py:53]:
```suggestion
    if source_language.lower() == language.lower():
```

**Review by COM8 — CHANGES_REQUESTED** [review]
Based on the DeepL api, the prefer uppercase language names.

**Comment by COM8** [comment]
Could you also update the documentation in the readme?

File: src/util.py (hunk 1 above)
Comment (Philipp000) [reply src/util.py]: I have updated the languages accordingly

**Comment by Philipp000** [comment]
I have now updated the languages for the DeepL API
And also included short sections for the translated dish titles in the documentation.

Is there anything else to do?

General question: How are these change requests generally handled? Shall I just click resolve, when I have implemented them? Or write a short answer? Or is the one who created them typically the one, who resolves them?

**Review by COM8 — APPROVED** [review]
Looks good now. Thanks!
Please rebase once again.

**Comment by COM8** [comment]
> General question: How are these change requests generally handled? Shall I just click resolve, when I have implemented them? Or write a short answer? Or is the one who created them typically the one, who resolves them?

Typically the one who resolves them closes them since they will be marked as outdated if something changes anyway.

**Comment by Philipp000** [comment]
It is now rebased, and should be ready for a merge

**Review by COM8 — APPROVED** [review]

**Pull request merged**